File: tests/adapters/scala/goldens/function_bodies/max_tokens_trim.scala
/**
 * Scala module for testing function body optimization.
 */

package com.example.test

trait UserRepository {
  def findById(id: Long): Option[User]
  def save(user: User): User
}

case class User(
  id: Long,
  name: String,
  email: String
)

class Calculator(val name: String = "default") {
  private val history: scala.collection.mutable.ListBuffer[String] =
    scala.collection.mutable.ListBuffer.empty

  def add(a: Int, b: Int): Int = {
    val result = a + b
    // … method body truncated (3 lines)
  }

  def multiply(a: Int, b: Int): Int = {
    val result = a * b
    // … method body truncated (2 lines)
  }

  def getHistory: List[String] = history.toList

  private def validateInput(value: Int): Boolean = {
    if (!value.toString.matches("^-?\\d+$")) {
    // … method body truncated (6 lines)
  }
}

case class ProcessingResult(
  valid: List[User],
  invalid: List[User]
)

object UserProcessor {
  def processUserData(users: List[User]): ProcessingResult = {
    val (valid, invalid) = users.partition { user =>
      user.id > 0 &&
    // … method body truncated (4 lines)
  }
}

// Function with pattern matching
def processValue(value: Any): String = value match {
  case s: String => s.toUpperCase
// … function body truncated (4 lines)

// Higher-order function
def processArray[T](items: List[T])(processor: T => T): List[T] = {
  items.map { item =>
    try {
      processor(item)
    } catch {
  // … function body truncated (5 lines)
}

// Implicit class (extension method)
implicit class StringOps(val s: String) extends AnyVal {
  def toTitleCase: String = {
    s.split(" ")
      .map(_.toLowerCase.capitalize)
      .mkString(" ")
  }
}

// Object with apply method
object Main extends App {
  val calc = new Calculator("test")
  println(calc.add(2, 3))
  println(calc.multiply(4, 5))

  val users = List(
    User(1, "Alice", "alice@example.com"),
    User(2, "Bob", "bob@example.com")
  )

  val result = UserProcessor.processUserData(users)
  println(s"Valid: ${result.valid}")
}

// Companion object
class Service private(val config: String) {
  def process(): Unit = {
    println(s"Processing with config: $config")
  }
}

object Service {
  def apply(config: String): Service = new Service(config)

  def default(): Service = new Service("default-config")
}

File: tests/adapters/java/goldens/function_bodies/max_tokens_trim.java
/**
 * Java module for testing function body optimization.
 */

package com.example.test;

import java.util.List;
import java.util.ArrayList;

public interface UserRepository {
    User findById(long id);
    User save(User user);
}

public class User {
    private final long id;
    private final String name;
    private final String email;

    public User(long id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }

    public long getId() { return id; }
    public String getName() { return name; }
    public String getEmail() { return email; }
}

public class Calculator {
    private final List<String> history;
    private final String name;

    public Calculator() {
        this("default");
    }

    public Calculator(String name) {
        this.name = name;
        this.history = new ArrayList<>();
    }

    public int add(int a, int b) {
        int result = a + b;
        // … method body truncated (2 lines)
        return result;
    }

    public int multiply(int a, int b) {
        int result = a * b;
        // … method body truncated
        return result;
    }

    public List<String> getHistory() {
        return new ArrayList<>(history);
    }

    private boolean validateInput(int value) {
        if (!String.valueOf(value).matches("^-?\\d+$")) {
        // … method body truncated (5 lines)

        return true;
    }
}

public class ProcessingResult {
    private final List<User> valid;
    private final List<User> invalid;

    public ProcessingResult(List<User> valid, List<User> invalid) {
        this.valid = valid;
        this.invalid = invalid;
    }

    public List<User> getValid() { return valid; }
    public List<User> getInvalid() { return invalid; }
}

public class UserProcessor {
    public static ProcessingResult processUserData(List<User> users) {
        // … method body truncated (11 lines)

        return new ProcessingResult(valid, invalid);
    }
}

// Generic function
public class ArrayProcessor {
    public static <T> List<T> processArray(List<T> items, Processor<T> processor) {
        List<T> result = new ArrayList<>();

        // … method body truncated (8 lines)

        return result;
    }
}

@FunctionalInterface
interface Processor<T> {
    T process(T item) throws Exception;
}

// Default export function equivalent
public class Main {
    public static void main(String[] args) {
        Calculator calc = new Calculator("test");
        // … method body truncated (8 lines)
    }
}
